File: backend/app/api/health.py
"""
Health check и мониторинг endpoints
"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from sqlalchemy import text
from app.core.database import get_db
from pydantic import BaseModel
from datetime import datetime
import redis
from app.core.config import settings

router = APIRouter(prefix="/health", tags=["health"])


class HealthResponse(BaseModel):
    status: str
    timestamp: datetime
    database: str
    redis: str
    version: str = "1.0.0"


@router.get("", response_model=HealthResponse)
async def health_check(db: Session = Depends(get_db)):
    """
    Проверка здоровья приложения
    
    Проверяет:
    - Доступность API
    - Подключение к базе данных
    - Подключение к Redis
    """
    
    # Проверка базы данных
    try:
        db.execute(text("SELECT 1"))
        db_status = "ok"
    except Exception as e:
        db_status = f"error: {str(e)}"
    
    # Проверка Redis
    try:
        r = redis.from_url(settings.REDIS_URL, decode_responses=True)
        r.ping()
        redis_status = "ok"
    except Exception as e:
        redis_status = f"error: {str(e)}"
    
    # Общий статус
    overall_status = "healthy" if db_status == "ok" and redis_status == "ok" else "unhealthy"
    
    return HealthResponse(
        status=overall_status,
        timestamp=datetime.utcnow(),
        database=db_status,
        redis=redis_status
    )


@router.get("/ping")
async def ping():
    """Простая проверка доступности"""
    return {"status": "ok", "message": "pong"}

File: backend-node/src/api/health.ts
import { Router, Request, Response } from 'express';
import prisma from '../core/database';
import { createClient } from 'redis';
import { config } from '../core/config';

const router = Router();

// Простая проверка
router.get('/ping', (req: Request, res: Response) => {
  res.json({ status: 'ok', message: 'pong' });
});

// Полная проверка здоровья
router.get('', async (req: Request, res: Response) => {
  let dbStatus = 'ok';
  let redisStatus = 'ok';

  // Проверка базы данных
  try {
    await prisma.$queryRaw`SELECT 1`;
  } catch (error) {
    dbStatus = `error: ${error instanceof Error ? error.message : 'unknown'}`;
  }

  // Проверка Redis
  let redisClient: ReturnType<typeof createClient> | null = null;
  try {
    redisClient = createClient({ url: config.redisUrl });
    await redisClient.connect();
    await redisClient.ping();
  } catch (error) {
    redisStatus = `error: ${error instanceof Error ? error.message : 'unknown'}`;
  } finally {
    if (redisClient) {
      try {
        await redisClient.quit();
      } catch (error) {
        // Игнорируем ошибки при закрытии
      }
    }
  }

  const overallStatus = dbStatus === 'ok' && redisStatus === 'ok' ? 'healthy' : 'unhealthy';

  res.json({
    status: overallStatus,
    timestamp: new Date().toISOString(),
    database: dbStatus,
    redis: redisStatus,
    version: '1.0.0',
  });
});

export default router;
